### DISCUSSION TITLE
Biome map looks strange with some datapacks at the Surface height

### DISCUSSION THREAD

**Discussion by Apollounknowndev** [jacobsjo/snowcapped#14] [Q&A] [answered]
[tectonic.zip](https://github.com/jacobsjo/snowcapped/files/10015868/tectonic.zip)
This is what it looks like at 319 height:
![image](https://user-images.githubusercontent.com/102649729/202015509-7a65483f-eee8-413a-b584-2277992952d6.png)
This is what it looks like at Surface height:
![image](https://user-images.githubusercontent.com/102649729/202015587-d5434a76-7983-4c3a-8eae-4d8c17af83bb.png)

**Comment by jacobsjo** [comment]
The zip you liked includes an assets folder which I guess causes some issues, but that is a different issue. (jacobsjo/mc-datapack-loader#1)

Apart from that I can't reproduce this issue, could you specify your browser and OS please?

**Comment by Apollounknowndev** [comment]
The OS I'm using is Windows 10. The browser I'm using is Google Chrome, but it seems to be an issue with all Chromium browsers

**Comment by jacobsjo** [comment]
On further investigation: some of your input density functions are not 2d, related to underground rivers? For that to work you need to uncheck the "2d input" checkbox at the top of the map. Does that fix the issue?

**Comment by Apollounknowndev** [comment]
Unchecking the 2d input checkbox does not fix it.
![image](https://user-images.githubusercontent.com/102649729/202022223-ea7625e0-2a63-46d4-be68-930397d0c83f.png)

**Comment by jacobsjo** [comment, marked as answer]
Ok, I see now. It is indeed different in chrome versus firefox. However, I think the way chrome does it is correct:
- There is no easy way for Snowcapped to determine the height of the surface. 
- Therefore, you need to create a special density function to specify the y level that the surface is at.
- These are named based on the name of the noise settings, in your case you can use `minecraft:overworld/snowcapped_surface`
- You can find the build in density functions here: https://github.com/jacobsjo/snowcapped/tree/main/vanilla_datapack_base/data/minecraft/worldgen/density_function

(I'll convert this issue to a discussion as I do think the reported behavior is correct)

**Comment by jacobsjo** [comment]
(#15 tracks why this is different in firefox)
